reference/client/get-pages.mdx

```diff
@@ -0,0 +1,366 @@
+  Used to either retrieve the published version or working version of Makeswift
+  pages. Can either be `Live` or `Working` 
+</ParamField>
+
+<ParamField query="includeOffline" type="boolean" default={false}>
+  Whether the results should include pages that are offline in Makeswift.
+</ParamField>
+
+## Return type
+
+```ts
+type MakeswiftPage = {
+  id: string,
+  path: string,
+  title: string | null,
+  description: string | null,
+  canonicalUrl: string | null,
+  socialImageUrl: string | null,
+  sitemapPriority: number | null,
+  sitemapFrequency: string | null,
+  createdAt: string,
+  updatedAt: string,
+  publishedAt: string | null,
+  isOnline: boolean | null,
+  excludedFromSearch: boolean | null,
+  locale: string,
+  localizedVariants: {
+    locale: string,
+    path: string
+  }[]
+}
+
+// Final return type
+type MakeswiftGetPagesResult = {
+  data: MakeswiftPage[],
+  hasMore: boolean
+}
+```
+
+## Concepts
+
+### Pagination & Async Iterators
+
+Pagination will only retrieve a portion of the total pages per request. If you
+have a large number of pages, you may need to paginate through the results to
+retrieve all of them.
+
+To make this easier, we've implemented the return type of `getPages` as an
+extension of a `Promise` and an async iterator. This abstraction enables you to
+consume the pages in multiple ways, enabling you to easily retrieve all pages
+without worrying about the implementation details of pagination:
+
+```ts get-pages.ts
+import { MakeswiftPage } from "@makeswift/runtime/next"
+import { client } from "makeswift/client"
+
+export async function getAllPages() {
+  // The `toArray` method on the iterator will return all pages as an array
+  const pagesFromToArray = await client.getPages().toArray()
+
+  // You can also get all pages with a `for await` loop
+  const pagesFromForAwait: MakeswiftPage[] = []
+  for await (const page of client.getPages()) {
+    pages.push(page)
+  }
+
+  // Get a single paginated result set 
+  const { data: pagesFromPagination } = await client.getPages({ limit: 5 })
+}
+```
+
+The custom async iterator returned by `getPages` includes the following methods:
+
+- `toArray()`: Consumes all pages in the iterator and returns them as an array.
+
+- `map(callback: (page: MakeswiftPage) => T)`: Maps over each page in the iterator to yield a new data
+  type. Returns another async iterator.
+
+- `filter(predicate: () => Boolean)`: Selectively yields pages based on a
+  predicate function. Returns another async iterator.
+
+Because `.map` and `.filter` return another async iterator, you can chain these
+methods together:
+
+```ts
+const pages = await client.getPages()
+  .filter((page) => !page.excludedFromSearch)
+  .map((page) => ({ id: page.id, title: page.title }))
+  .toArray()
+```
+
+For more details on async iterators, see the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator).
```

Comment:
Async generators might throw people off a bit since this isn't necessarily what we're trying to present to people here. Instead, it's about async iterables and the utilities we created: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols